Dialect api

VitGottwald opened the pull request.

VitGottwald committed:
refactor: remove extra function

VitGottwald committed:
refactor: do not wait for dialect  activation

VitGottwald committed:
refactor: return function to unregister dialect

VitGottwald committed:
refactor: update package-lock.json formatting

by running `npm install`

Probably because we used to have .ediotorconfig,
npm used to format the package-lock.json by 4 spaces

VitGottwald committed:
refactor: use unregister dialect function

VitGottwald committed:
refactor: use extension context

to get extension id and extension path

VitGottwald committed:
refactor: call variable by what it meanins

VitGottwald committed:
refactor: use vscode.Uri.joinPath istead path.join

VitGottwald committed:
refactor: get extension id from extension context

VitGottwald committed:
refactor: remove redundant type annotation

VitGottwald committed:
chore: add note to include jar file name in URI

VitGottwald committed:
refactor: use URIs instead of path in dialectAPI

VitGottwald committed:
refactor: move API checks into separate module

VitGottwald committed:
refactor: move dialect API to separate package

VitGottwald committed:
refactor: [DaCo] use cobol-dialect-api package

this required unifing the version of @types/vscode and typescript
so that the code would compile

VitGottwald committed:
chore: add vscode setup for debugging daco ext

VitGottwald committed:
chore: add todo

VitGottwald committed:
build: build dialect api

VitGottwald committed:
deps: upgrade @vscode/vsce

ishche commented on clients/cobol-dialect-api/src/index.ts:
```diff
@@ -0,0 +1,32 @@
+  (): void;
+}
+
+export const registerDialect = async (
```
I think it makes sense to add a simple test for this.

VitGottwald replied:
I will do this in next iteration that will include refactoring of the API exported by COBOL LS extension. The focus of this PR is on the clean up of the API faced by dialect extender.

VitGottwald committed:
build: explicitly declare what to package

VitGottwald committed:
chore: add config for prettier on clients level

if developer opens the `clients` folder so they can edit multiple
extensions at the same time, they should get the formatting right
(by prettier)

ishche commented on clients/.vscode/extensions.json:
```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2022 Broadcom.
```
it's 2023 already :)

ishche commented on clients/daco-dialect-support/.vscodeignore:
```diff
@@ -0,0 +1,25 @@
+# Ignore everything
```
copyright header is missing

VitGottwald committed:
chore: add prettier config for vscode

VitGottwald committed:
Add/fix copyright headers

VitGottwald committed:
build: add dialect api build step

VitGottwald committed:
chore: fix copyright date
